test/rss.cpp

```diff
@@ -166,3 +166,43 @@ TEST_CASE("W3CDTF parser behaves correctly") {
 		REQUIRE(rsspp::rss_parser::__w3cdtf_to_rfc822("") == "");
 	}
 }
+
+TEST_CASE("W3CDTF behaves correctly with different local timezones") {
+	// There has been a problem in the C date conversion functions when the TZ is
+	// set to different locations, and localtime is in daylight savings. One of
+	// these two next tests sections should be in active daylight savings.
+	// https://github.com/akrennmair/newsbeuter/issues/369
+	
+	char *tz = getenv("TZ");
```

Comment:
My manpage on `getenv` says:

> The string pointed to by the return value of getenv() may be statically allocated, and can be modified by a subsequent call to getenv(), putenv(3), setenv(3), or unsetenv(3).

So I'd rather you immediately copy the value to a safe place that you allocated yourself. Stack-allocated array of large enough size should do just fine, I guess.